bids/layout/index.py

```diff
@@ -31,12 +31,12 @@ def _check_path_matches_patterns(path, patterns):
     """Check if the path matches at least one of the provided patterns. """
     if not patterns:
         return False
-    path = os.path.abspath(path)
+    path = path.absolute()
     for patt in patterns:
-        if isinstance(patt, str):
+        if isinstance(patt, Path):
             if path == patt:
                 return True
-        elif patt.search(path):
+        elif patt.search(str(path)):
```

Comment:
Not sure atm what to expect for `patt` to be here. Please someone check.

Reply:
`patt` is supposed to be string or regex. So this is the regex case.

Reply:
Just to be clear: the `path` is always of type `Path` here, so the conversion to `str` is necessary.